### DISCUSSION TITLE
lossless-cut requires xwayland?

### DISCUSSION THREAD

**Discussion by rieje** [mifi/lossless-cut#1675] [Q&A]
I am getting the error on the latest version 3.54:

    [43563:0805/165903.394342:ERROR:ozone_platform_x11.cc(239)] Missing X server or $DISPLAY
    [43563:0805/165903.394371:ERROR:env.cc(255)] The platform failed to initialize.  Exiting.
    The futex facility returned an unexpected error code.
    zsh: IOT instruction (core dumped)  losslesscut

With xwayland disabled. Is it possible for lossless-cut to support pure Wayland?

**Comment by mifi** [comment]
nothing in losslesscut uses xwayland. it all depends on [electron's wayland support](https://github.com/search?q=repo%3Aelectron%2Felectron+wayland&type=issues), which i don't know about. also https://www.reddit.com/r/archlinux/comments/zpuey0/electron_apps_wont_run_on_wayland/

**Comment by chriad** [comment]
```
flatpak list|grep losslesscut
LosslessCut	no.mifi.losslesscut	3.55.1	stable	flathub	system
```

I had problems with drag-and-drop with xwayland-compatible losslesscut when I recently changed to a new desktop environment.

You can run:

`$ /var/lib/flatpak/app/no.mifi.losslesscut/x86_64/stable/active/files/lossless-cut/losslesscut --no-sandbox --ozone-platform-hint=auto`

Maybe the maintainer can add `--ozone-platform-hint=auto` to `/var/lib/flatpak/app/no.mifi.losslesscut/x86_64/stable/active/files/bin/run.sh` so that it will run on wayland when this is available?

`--ozone-platform=wayland` also works.

I don't see how the user could add this flag in e.g. `/var/lib/flatpak/exports/share/applications/no.mifi.losslesscut.desktop` or in a configuration file someplace?

**Reply by mifi** [reply]
flatpak is not maintained by me, but you can try here: https://github.com/flathub/no.mifi.losslesscut
